### PR TITLE
TYP: MergeHow + JoinHow

### LINKED ISSUE #48503: ENH: `.merge` - specify string literals for `.merge` `how` keyword
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

The `how` keyword for `DataFrame.merge` is typed as a string - which means that type checkers can miss potential errors, as it only accepts a subset of strings ("inner", "outer", "left", "right", "cross").

### Feature Description

`pandas._typing` : add a `MergeHow` type, specifying the string literals that `how` accepts

`pandas.core.frame`: `how` as accepting an argument of type `MergeHow`

### Alternative Solutions

Alternatively we could specify the string literals in the `merge` method function header

### Additional Context

[Related PR for `pandas-stubs`](https://github.com/pandas-dev/pandas-stubs/pull/289#issuecomment-1242756941)

### PR DESCRIPTION
**Pull request by debnathshoham**
- [x] closes #48503 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/_typing.py [hunk 1]
```diff
@@ -343,6 +343,12 @@ def closed(self) -> bool:
 # dropna
 AnyAll = Literal["any", "all"]
 
+# merge
+MergeHow = Literal["left", "right", "inner", "outer", "cross"]
+
+# join
+JoinHow = Literal["left", "right", "inner", "outer"]
```
Comment (phofl) [inline pandas/_typing.py:350]: Join supports cross as well

Comment (debnathshoham) [reply pandas/_typing.py:350]: typed join with `MergeHow`.. used `JoinHow` for the indices.

I actually took this from pandas-stubs

https://github.com/pandas-dev/pandas-stubs/blob/db3b883dbe8d22863c8aca37c89431afce3b57e2/pandas-stubs/_typing.pyi#L299-L300

Comment (phofl) [reply pandas/_typing.py:350]: Could you check how they type DataFrame.join?

Comment (phofl) [reply pandas/_typing.py:350]: Yep

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @debnathshoham

**Pull request merged**